scripts/ndjson_to_csv/ndjson_to_csv.py

```diff
@@ -0,0 +1,74 @@
+# /usr/bin/python3
+
+import datetime
+import json
+
+from functools import reduce
+
+from tqdm import tqdm
+import pandas as pd
+
+def flatten(data, parent_key='', sep='.'):
+    items = []
+    for k, v in data.items():
+        new_key = parent_key + sep + k if parent_key else k
+        if isinstance(v, dict):
+            items.extend(flatten(v, new_key, sep).items())
+        elif isinstance(v, list):
+            if not isinstance(v[0], (str, int)):
+                if len(v) > 1:
+                    for index, item in zip(range(len(v)), v):
+                        if index > 0:
+                            items.extend(flatten(
+                                item,
+                                '{}{}'.format(new_key, index+1),
+                                sep
+                            ).items())
+                        else:
+                            items.extend(flatten(item, new_key, sep).items())
+                else:
+                    for item in v:
+                        items.extend(flatten(item, new_key, sep).items())
+            else:
+                list_strings = [str(item) for item in v]
+                items.append((new_key, ' '.join(list_strings)))
+        else:
+            items.append((new_key, v))
+    return dict(items)
+
+
+if __name__=='__main__':
+    print("This tool can convert FHIR data stored in an NDJSON file to a CSV format.")
+    while True:
+        # Ingest the report results
+        filename = input("Enter the name of the NDJSON file you wish to convert: ")
+        f = open(filename, 'r')
+        
+        # Format data
+        string_json = f.read()
```

Comment:
what size NDJSON files do you expect users to be able to use this script with?  I've heard that some of the files ACO's produce are GBs in size, reading this in all at once might cause memory errors

Reply:
That's a really good point that I hadn't taken into account.

Reply:
@ericbuckley I've completely refactored the way that this works.

Instead of reading the source into memory, I'm iterating over the file line by line. This _should not_ keep too many things in memory, however, I am using a tempfile that will be roughly the same size as the source while the script is running. Let me know if you want to discuss how that may or may not impact folks.